pkg/chart/chart.go

```diff
@@ -338,3 +352,42 @@ func (c *ChartClient) GetChart(details *Details, netClient kube.HTTPClient, requ
 
 	return chart, nil
 }
+
+// RegistrySecretsPerDomain checks the app repo and available secrets
+// to return the secret names per registry domain.
+//
+// These are actually calculated during InitNetClient when we already have a
+// k8s client with the user token.
+func (c *ChartClient) RegistrySecretsPerDomain() map[string]string {
+	return c.registrySecretsPerDomain
+}
+
+func getRegistrySecretsPerDomain(appRepoSecrets []string, namespace string, client kube.Handler) (map[string]string, error) {
```

Comment:
unless I am missing something, you would need to return here `client kube.AuthHandler`. The `handler` is intentionally not exported so we don't mistakenly use it without setting it up with a token first (because if you don't do so, it would use the backend serviceaccount token if I recall correctly).

Reply:
`client kube.AuthHandler` is an interface defining the `AsUser(token)` method, so yes, I could pass that as well as the token to this method so we're explicit. It'd be nicer to explicitly have an interface `kube.UserHandler` to force the client here to be the result of `AsUser(token)`, but I'll leave that for another day, and just update to use `AuthHandler` and pass in the token. Thanks.